src/Renderer/Utils/Array.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+#include <memory>
+#include <iostream>
+#include <cstdint>
+
+namespace SnekVk::Utils
+{
+    template<typename Array>
+    class ArrayIterator
+    {
+        public: 
+            using ValueType = typename Array::ValueType;
+            using PointerType = ValueType*;
+            using ReferenceType = ValueType&;
+        public:
+
+        ArrayIterator(PointerType pointer) 
+            : pointer{pointer} {};
+        
+        ArrayIterator& operator++() 
+        {
+            pointer++;
+            return *this;
+        }
+
+        ArrayIterator operator++(int)
+        {
+            ArrayIterator iterator = *this;
+            ++(*this);
+            return iterator;
+        }
+
+        ArrayIterator& operator--() 
+        {
+            pointer--;
+            return *this;
+        }
+
+        ArrayIterator operator--(int)
+        {
+            ArrayIterator iterator = *this;
+            --(*this);
+            return iterator;
+        }
+
+        ReferenceType operator[](size_t index)
+        {
+            return *(pointer + index);
+        }
+
+        PointerType operator->()
+        {
+            return pointer;
+        }
+
+        ReferenceType operator*()
+        {
+            return *pointer;
+        }
+
+        bool operator==(const ArrayIterator& other) const
+        {
+            return pointer == other.pointer;
+        }
+
+        bool operator!=(const ArrayIterator& other) const
+        {
+            return !(*this == other);
+        }
+
+        private:
+        
+        PointerType pointer;
+    };
+
+    template<typename T>
+    class Array 
+    {
+        public:
+        using ValueType = T;
+        using Iterator = ArrayIterator<Array>;
+        public:
+
+        Array() : size{0} {};
+
+        Array(size_t size) : size{size} 
+        {
+            data = new T[size];
+        }
+
+        Array(uint32_t size) : size{static_cast<size_t>(size)} 
+        {
+            data = new T[this->size];
+        }
+
+        Array(std::initializer_list<T> values)
+        {
+            data = new T[values.size()];
+            
+            size_t index = 0;
+            for (auto val : values)
+            {
+                data[index] = val;
+                index++;
+            }
+
+            size = values.size();
+        }
+
+        Array(const Array<T>& other)
+        {
+            if (data) delete [] data;
+
+            Copy(other);
+        }
```

Comment:
`data` is initialised to nullptr by default
```suggestion
        Array(const Array<T>& other)
        {
            Copy(other);
        }
```